### PR TITLE
Add CLI command to validate scenario data file from definitions

### LINKED ISSUE #404: CLI to validate a scenario data file
To make it easier for non-expert users to check their data against a DataStructureDefinition, we need a simple CLI that takes an input file (IAMC scenario data) and validates it against a the definitions.

Basically, this should be a reduced version of `cli_run_workflow`, but only take an `input file` (required) and  a `definitions` folder (default to 'definitions').

### PR DESCRIPTION
**Pull request by dc-almeida**
First step towards #404

### CODE REVIEW

**Review by phackstock — CHANGES_REQUESTED** [review]
Nice addition @dc-almeida (sorry for jumping in here with a review).
Couple of comments in line below.

**Comment by dc-almeida** [comment]
Thanks for the hawk-eye as always!

**Comment by phackstock** [comment]
> Thanks for the hawk-eye as always!

You're welcome, always easier to spot these things in other people's work :D

**Review by phackstock — APPROVED** [review]
Thanks for the quick changes @dc-almeida.
Good to be merged from my side, in principle.
However, as mentioned in #419, I think the name of the command might be misleading and I think it should include an option to specify the dimensions that should be validated.

File: tests/data/cli/validate_scenarios/definitions/region/regions.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- common:
```
Comment (danielhuppmann) [inline tests/data/cli/validate_scenarios/definitions/region/regions.yaml:1]: Can you use an existing definitions-folder in the test-data? We're adding quite a lot of very similar items here and it's hard to keep an overview...

File: nomenclature/cli.py [hunk 2]
```diff
@@ -277,3 +277,30 @@ def cli_run_workflow(
...
+@click.argument("input_file", type=click.Path(exists=True, path_type=Path))
+@click.option(
+    "--definitions",
+    help="Optional name for definitions folder",
+    type=click.Path(exists=True, path_type=Path),
+    default="definitions",
+)
+def cli_validate_scenarios(input_file: Path, definitions: Path):
+    """Run a given input file through a DataStructureDefinition validation,
+    checking it against defined codelist(s)
```
Comment (danielhuppmann) [inline nomenclature/cli.py:292]: Try to keep the docstring such that the first line is really only one line, per numpydoc standards.
```suggestion
    """Validate a scenario file against the codelists of a project
```

**Review by danielhuppmann — APPROVED** [review]
Thank you!

**Comment by phackstock** [comment]
Before we go ahead with the merge if I understood our discussion in #419 correctly, this PR is still missing a dimensions option so that you can specify which dimension you want to validate, correct @danielhuppmann?

**Comment by danielhuppmann** [comment]
I have a suggestion: we merge as is, and then @dc-almeida does a follow-up PR adding the dimensions for this CLI, and also extending the tests for the `validate-scenario` and `validate-project` to make sure that all three cases work as expected:
- explicit `dimensions` argument (this is currently tested in `test_cli_custom_dimensions_runs()`)
- if not given, use `dimensions` argument set from `nomenclature.yaml` (not sure if this is tested)
- if also not given, check all available folders

**Comment by phackstock** [comment]
Sounds good to me.

**Pull request merged**